src/Stripe.net/Infrastructure/FormEncoding/FormEncoder.cs

```diff
@@ -16,60 +18,53 @@ namespace Stripe.Infrastructure.FormEncoding
     /// </summary>
     internal static class FormEncoder
     {
-        /// <summary>Creates the HTTP query string for a given options class.</summary>
-        /// <param name="options">The options class for which to create the query string.</param>
-        /// <returns>The query string.</returns>
-        public static string EncodeOptions(BaseOptions options)
-        {
-            return EncodeValue(options, null);
-        }
-
-        /// <summary>
-        /// Join several query strings together, omitting <c>null</c> or empty strings.
-        /// </summary>
-        /// <param name="queries">One or more query strings to be joined.</param>
-        /// <returns>The joined query string.</returns>
-        public static string JoinQueries(params string[] queries)
+        /// <summary>Creates an <see cref="HttpContent"/> for a given options class.</summary>
+        /// <param name="options">The options class.</param>
+        /// <returns>The <see cref="HttpContent"/>.</returns>
+        public static HttpContent CreateHttpContent(BaseOptions options)
         {
-            return string.Join("&", queries.Where(q => !string.IsNullOrEmpty(q)));
-        }
+            // If options is null, we create an empty FormUrlEncodedContent because we still
+            // want to send the Content-Type header.
+            if (options == null)
+            {
+                return new FormUrlEncodedContent(new List<KeyValuePair<string, string>>());
+            }
 
-        /// <summary>Append one or more query strings to a URL.</summary>
-        /// <param name="url">The base URL.</param>
-        /// <param name="queries">One or more query strings to be appended to the URL.</param>
-        /// <returns>The full URL with all the query strings.</returns>
-        public static string AppendQueries(string url, params string[] queries)
-        {
-            var fullQuery = JoinQueries(queries);
+            var flatParams = FlattenParamsValue(options, null);
 
-            if (!string.IsNullOrEmpty(fullQuery))
+            if (flatParams.All(kvp => kvp.Value is string))
```

Comment:
Is the idea here that we know that by this point all parameters will have for sure been converted to a string except for those intended for multipart encoding? If so, mind adding a comment to that effect?

Reply:
Yes, that's it. The values in the flattened parameters map are either strings or streams. If all of them are strings, then the content can be encoded as `application/x-www-form-urlencoded`, otherwise it needs to be encoded as `multipart/form-data`. I'll add a comment 👍